pkg/cache/scheduler/fair_sharing.go

```diff
@@ -168,9 +166,16 @@ func calculateLendable(node hierarchicalResourceNode) map[corev1.ResourceName]in
 
 // parseFairWeight parses FairSharing.Weight if it exists,
 // or otherwise returns the default value of 1.
-func parseFairWeight(fs *kueue.FairSharing) resource.Quantity {
+func parseFairWeight(fs *kueue.FairSharing) float64 {
 	if fs == nil || fs.Weight == nil {
-		return oneQuantity
+		return defaultWeight
 	}
-	return *fs.Weight
+
+	// we make a deep copy to avoid any data race, as this is a
+	// mutating method.  even though parseFairWeight only called
```

Comment:
```suggestion
	// mutating method even though parseFairWeight only called
```

nit

Reply:
I will leave as is. Without the period, it seems closer to a run-on sentence to me